server/src/main/java/se/fortnox/reactivewizard/server/RxNettyServer.java

```diff
@@ -65,17 +75,50 @@ public void run() {
     }
 
     private void registerShutdownHook() {
-        Runtime.getRuntime().addShutdownHook(new Thread(() -> shutdownHook(config, server, connectionCounter)));
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> shutdownHook(config, server, eventLoopGroup, connectionCounter)));
+    }
+
+    public static void registerShutdownDependency(Runnable blockShutdownUntil) {
+        if (RxNettyServer.blockShutdownUntil != null && blockShutdownUntil != null) {
+            throw new IllegalStateException("Shutdown dependency is already registered");
```

Comment:
Perhaps you can have a list of Runnables so that we can have multiple shutdown hooks?

Reply:
postponed.